src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -0,0 +1,621 @@
+                 kfsVendorDataWrapper.setVendorDetail(null);
+            }
+            break;
+            
+            case PaymentWorksConstants.PaymentWorksNewVendorTaxBusinessRule.INDIVIDUAL_NOT_US_FOREIGN_TAX_CLASS_INDIVIDUAL: 
+            {    kfsVendorDataWrapper.getErrorMessages().add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.FOREIGN_VENDOR_PROCESSING_NOT_AUTOMATIC_YET);
+                 kfsVendorDataWrapper.setVendorDetail(null);
+            }
+            break;
+            
+            case PaymentWorksConstants.PaymentWorksNewVendorTaxBusinessRule.INDIVIDUAL_NOT_US_FOREIGN_TAX_CLASS_OTHER: 
+            {    kfsVendorDataWrapper.getErrorMessages().add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.FOREIGN_VENDOR_PROCESSING_NOT_AUTOMATIC_YET);
+                 kfsVendorDataWrapper.setVendorDetail(null);
+            }
+            break;
+            
+            case PaymentWorksConstants.PaymentWorksNewVendorTaxBusinessRule.NOT_INDIVIDUAL_NOT_US_EIN: 
+            {    kfsVendorDataWrapper.getErrorMessages().add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.FOREIGN_VENDOR_PROCESSING_NOT_AUTOMATIC_YET);
+                 kfsVendorDataWrapper.setVendorDetail(null);
+            }
+            break;
+            
+            case PaymentWorksConstants.PaymentWorksNewVendorTaxBusinessRule.NOT_INDIVIDUAL_NOT_US_FOREIGN: 
+            {    kfsVendorDataWrapper.getErrorMessages().add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.FOREIGN_VENDOR_PROCESSING_NOT_AUTOMATIC_YET);
+                 kfsVendorDataWrapper.setVendorDetail(null);
+            }
+            break;
+            
+            case PaymentWorksConstants.PaymentWorksNewVendorTaxBusinessRule.COULD_NOT_DETERMINE_TAX_RULE_TO_USE:
+            default: 
+            {
+                kfsVendorDataWrapper.getErrorMessages().add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.COULD_NOT_DETERMINE_TAX_BUSINESS_RULE_TO_USE);
+                kfsVendorDataWrapper.setVendorDetail(null);
+            }
+            break;
+        }
+        return kfsVendorDataWrapper;
+    }
+    
+    private String convertIsoCountryCodeToFipsCountryCode(String isoCountryCode, Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap) {
+        return (paymentWorksIsoToFipsCountryMap.get(isoCountryCode).get(0)).getFipsCountryCode();
+    }
+
+    private List<VendorAddress> buildVendorAddresses(PaymentWorksVendor pmwVendor, Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap) {
+        List<VendorAddress> allVendorAddresses = new ArrayList<VendorAddress>();
+        allVendorAddresses.add(buildTaxAddress(pmwVendor, paymentWorksIsoToFipsCountryMap));
+        allVendorAddresses.add(buildRemitAddress(pmwVendor, paymentWorksIsoToFipsCountryMap));
+        
+        // WHEN FIRST PO IMPLEMENTATION IS DONE ADD THIS CONDITIONAL BACK INTO CODE.
```

Comment:
we shouldn't have these kind of comments in the code. Perhaps we could find a better way to deal with this, like a method for PO that now does nothing?

Reply:
Removed comments